monitor.py

```diff
@@ -60,43 +60,47 @@ def get_jobs(prow_link):
-            if script_content:
-                if 'allBuilds' in script_content:
-                    selected_script_element = script_content
-                    break
+        if response.status_code == 200:
+            soup = BeautifulSoup(response.text, 'html.parser')
+            script_elements = soup.find_all('script')
+            selected_script_element = None
+
+            for script_element in script_elements:
+                script_content = script_element.string
+                if script_content:
+                    if 'allBuilds' in script_content:
+                        selected_script_element = script_content
+                        break
         
-        if selected_script_element:
-            var_name = 'allBuilds'
-            pattern = rf'{var_name}\s*=\s*(.*?);'
-
-            match = re.search(pattern, selected_script_element)
-            if match:
-                all_jobs=match.group(1)
-                #print(all_jobs)
-                try:
-                    all_jobs_parsed=json.loads(all_jobs)
-                    current_date=get_current_date()
-                    jobs_run_today = []
-                    for ele in all_jobs_parsed:
-                        job_time=parse_job_date(ele["Started"])
-                        if job_time == current_date and ele["Result"] != "PENDING":
-                            job_log_path = ele["SpyglassLink"]
-                            jobs_run_today.append(job_log_path)
-                    return jobs_run_today
-                except json.JSONDecodeError as e:
-                    return "Failed to extract the spy-links from spylink please check the UI!"
-                    
-    else:
-        return "Failed to get the prowCI response"
+            if selected_script_element:
+                var_name = 'allBuilds'
+                pattern = rf'{var_name}\s*=\s*(.*?);'
+
+                match = re.search(pattern, selected_script_element)
+                if match:
+                    all_jobs=match.group(1)
+                    #print(all_jobs)
+                    try:
+                        all_jobs_parsed=json.loads(all_jobs)
+                        current_date=get_current_date()
+                        jobs_run_today = []
+                        for ele in all_jobs_parsed:
+                            job_time=parse_job_date(ele["Started"])
+                            if job_time == current_date and ele["Result"] != "PENDING":
+                                job_log_path = ele["SpyglassLink"]
+                                jobs_run_today.append(job_log_path)
+                        return jobs_run_today
+                    except json.JSONDecodeError as e:
+                        return "Failed to extract the spy-links from spylink please check the UI!"                    
+        else:
+            return "Failed to get the prowCI response"
+    except requests.Timeout:
+        return "Request timed out"
```

Comment:
Can we add the exception details in the return statement to provide detailed info.

Reply:
sometimes the exception details can be huge and since we are using this code to send daily build updates this will result in changing the daily build updates message structure so instead I have added a brief message of those exceptions, for debugging purpose we can print the error object for more information.